### PR TITLE
[zos-console] Added method to get a command response

### LINKED ISSUE #78: [zos-console] Add method to get a command response
Add a method to the Python zos-console SDK to call the z/OSMF REST API `GET /zosmf/restconsoles/consoles/<console-name>/solmsgs/C<key-number>`

* z/OSMF REST API documentation: https://www.ibm.com/docs/en/zos/2.5.0?topic=services-get-command-response
* Python zos-console SDK in the "next" branch: https://github.com/zowe/zowe-client-python-sdk/tree/next/src/zos_console
* Node.js zos-console SDK for comparison: https://github.com/zowe/zowe-cli/blob/master/packages/zosconsole/src/CollectCommand.ts

### PR DESCRIPTION
**Pull request by aadityasinha-dotcom**
Added a method to call the z/OSMF REST API `GET /zosmf/restconsoles/consoles/<console-name>/solmsgs/C<key-number>`
Signed-off-by: Aaditya Sinha [aadityasinha009@gmail.com](mailto:aadityasinha009@gmail.com)
Resolves #78

### CODE REVIEW

**Comment by aadityasinha-dotcom** [comment]
@t1m0thyj can u review this

File: src/zos_console/zowe/zos_console_for_zowe_sdk/console.py [hunk 1]
```diff
@@ -43,3 +43,24 @@ def issue_command(self, command, console=None):
...
+            The command response key from the Issue Command request.
+        console
+            The console that should be used to get the command response.
+        Returns
+        -------
+        json
+            A JSON containing the response to the command
+        """
+        custom_args = self._create_custom_request_arguments()
+        request_url = "{}/solmsgs/{}".format(console or "defcn", response_key)
```
Comment (aadityasinha-dotcom) [inline src/zos_console/zowe/zos_console_for_zowe_sdk/console.py:63]: should I also create test for this?

Comment (t1m0thyj) [reply src/zos_console/zowe/zos_console_for_zowe_sdk/console.py:63]: Yes, please add unit and integration tests 🙂 They can be modelled after tests that you've created for previous PRs.

File: tests/unit/test_zos_console.py [hunk 2]
```diff
@@ -19,4 +20,12 @@ def setUp(self):
...
-        self.assertIsInstance(console, Console)
\ No newline at end of file
+        self.assertIsInstance(console, Console)
+    
+    @mock.patch('requests.Session.send')
+    def test_get_response_should_return_messages(self, mock_send_request):
+       """Getting z/OS Console response messages on sending a response key"""
+       mock_send_request.return_value = mock.Mock(headers={"Content-type": "application/json"}, status_codes=200)
+
+       Console(self.session_details).get_response(self)
```
Comment (aadityasinha-dotcom) [inline tests/unit/test_zos_console.py:30]: @t1m0thyj  added the unit test can you test that.

Comment (t1m0thyj) [reply tests/unit/test_zos_console.py:30]: Thanks, I will take a look 🙂 You should also be able to verify unit tests by running them locally with the command `pytest tests/unit` (no mainframe connection is needed, that is only for the integration tests).

Comment (t1m0thyj) [inline tests/unit/test_zos_console.py:30]: The response key should be a string like `"console-key"` instead of the class instance `self`.

Comment (aadityasinha-dotcom) [reply tests/unit/test_zos_console.py:30]: Hi yes there is some error
Show unexpectedStatus

**Review by t1m0thyj — CHANGES_REQUESTED** [review]
@aadityasinha-dotcom Please amend your last commit to fix the failing signoff - the "Signed-off-by" message needs to start on a new line.

Thanks for adding the unit test, left a few comments 🙂

File: tests/unit/test_zos_console.py (hunk 2 above)
Comment (aadityasinha-dotcom) [reply tests/unit/test_zos_console.py:30]: okay thanks

Comment (aadityasinha-dotcom) [reply tests/unit/test_zos_console.py:30]: thank you now the tests are passing

Comment (aadityasinha-dotcom) [reply tests/unit/test_zos_console.py:30]: also how can I amend the last commit?

Comment (t1m0thyj) [reply tests/unit/test_zos_console.py]: You should be able to amend the commit using your preferred Git tool, and then force push the changes to this branch.

I believe the Git CLI commands would be:
```
git commit --amend -m "changes" -s
git push -u origin get_command1 --force-with-lease
```

**Comment by zFernand0** [comment]
You may still need to amend this commit 77ea74d4ef6bf2156ebb792890157250ead20c85
Seems like dcobot needs signoffs to be in their own line 😢

**Comment by aadityasinha-dotcom** [comment]
@t1m0thyj sorry this PR has some signing off conflicts. Made a new PR https://github.com/zowe/zowe-client-python-sdk/pull/143 please review that :slightly_smiling_face:

**Pull request closed without merging**